### ISSUE TITLE
Calling request.body() may result in a NullPointerException

### ISSUE THREAD

**Issue by rbordon-meli** [perwendel/spark#1072]
As the title said, calling request.body() may result in a NullPointerException.

This may be caused because there is no response handle if an error occurred during readBodyAsBytes().
So the response of bodyAsBytes() is null, then invoking bytes.length in String class throws a NPE.

**Request**
```
public String body() {

    if (body == null) {
        body = StringUtils.toString(bodyAsBytes(), servletRequest.getCharacterEncoding());
    }

    return body;
}

public byte[] bodyAsBytes() {
    if (bodyAsBytes == null) {
        readBodyAsBytes();
    }
    return bodyAsBytes;
}

private void readBodyAsBytes() {
    try {
        bodyAsBytes = IOUtils.toByteArray(servletRequest.getInputStream());
    } catch (Exception e) {
        LOG.warn("Exception when reading body", e);
    }
}
```

**StringUtils**
```
public static String toString(byte[] bytes, String encoding) {
    String str;

    if (encoding != null && Charset.isSupported(encoding)) {
        try {
            str = new String(bytes, encoding);
        } catch (UnsupportedEncodingException e) {
            // Uses same func as Charset.isSupported (cannot happen)
            str = new String(bytes);
        }
    } else {
        str = new String(bytes);
    }

    return str;
}
```

**String**
```
public String(byte bytes[], String charsetName)
        throws UnsupportedEncodingException {
    this(bytes, 0, bytes.length, charsetName);
}
```

A way to reproduce the bug, is debugging the app with a breakpoint in the line
`bodyAsBytes = IOUtils.toByteArray(servletRequest.getInputStream());`
Then cancel the request, in order to produce an IOException. This makes bodyAsBytes null.
I´ve used a JSON request of 1.5 MB

**Comment by emportella** [comment]
I`m currently suffering intensive with this bug. about 10 times per sec in multiple endpoins for both PUT and POST

**Comment by robax** [comment]
+1 also running into a problem related to this.

**Comment by sanket-sheth** [comment]
Any update on this issue?

**Comment by perwendel** [comment]
Fixed in #1170

**Comment by robax** [comment]
It seems this patch isn't in 2.9.2. Can we get a quick build with this patch? @perwendel 
And thank you!
